lib/validator/formchecker.cpp

```diff
@@ -492,6 +583,28 @@ Expect<void> FormChecker::checkInstr(const AST::Instruction &Instr) {
     }
     return unreachable();
   }
+  case OpCode::Return_call_ref: {
+    auto TypeIdx = Instr.getTargetIndex();
+    if (TypeIdx >= Types.size()) {
+      // Call function index out of range
+      spdlog::error(ErrCode::Value::InvalidFuncIdx);
+      spdlog::error(ErrInfo::InfoForbidIndex(
+          ErrInfo::IndexCategory::FunctionType, TypeIdx,
+          static_cast<uint32_t>(Types.size())));
+      return Unexpect(ErrCode::Value::InvalidFuncIdx);
+    }
+    if (!match_type(Types[TypeIdx].second, Returns)) {
+      spdlog::error(ErrCode::Value::TypeCheckFailed);
+      // TODO: Print the error info of types.
```

Comment:
Use `ErrInfo::InfoMismatch` instead of this TODO comment.

Reply:
fixed.